src/test/java/SeleniumWebDriverTest.java

```diff
@@ -0,0 +1,47 @@
+
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+public class SeleniumWebDriverTest {
+        @Test
+        public void testDriver() throws InterruptedException {
+            System.setProperty("webdriver.chrome.driver", "src/main/resources/chromedriver.exe");
+            WebDriver driver = new ChromeDriver();
+            driver.manage().window().maximize();
+            driver.get("https://www.selenium.dev");
+            Thread.sleep(3000);
+            driver.findElement(By.xpath("//a[@class='nav-item' and normalize-space()='Downloads']")).click();
+            String stableVersion = "3.141.59";
+            WebElement actualElement = driver.findElement(By.xpath("//a[@href=\"https://selenium-release.storage.googleapis.com/3.141/selenium-server-standalone-3.141.59.jar\" and text()='3.141.59']"));
+            String actualName = actualElement.getText();
+            Assert.assertEquals(actualName, stableVersion, "Verify the latest stable version is 3.141.59!");
+            WebElement searchElement = driver.findElement(By.xpath("//input[@name='search']"));
+            String searchName = "selenium webdriver";
+            searchElement.sendKeys(searchName + Keys.ENTER);
+            Thread.sleep(5000);
+            List<WebElement> links = driver.findElements(By.tagName("a"));
+            for (WebElement element : links) {
+                if (element.getText().contains("Selenium WebDriver")) {
+                    System.out.println(element.getText());
+                    String linkName = element.getText().toLowerCase();
+                    Assert.assertEquals(linkName, searchName, "Verify the found link contains our desired phrase!");
```

Comment:
linkName isn't equal to searchName and your test passes accidently, because the link that contains "selenium webdriver" doesn't have other text in it. But if link has a text like this "selenium webdriver {someText}", your test would fail.
So change the logic inside loop.